DataFormats/MuonReco/interface/MuonSegmentMatch.h

```diff
@@ -46,7 +48,8 @@ namespace reco {
 
          DTRecSegment4DRef  dtSegmentRef;
          CSCSegmentRef      cscSegmentRef;
-
+	 GEMSegmentRef      gemSegmentRef;
+	 ME0SegmentRef      me0SegmentRef;

```

Comment:
this, together with the use case pattern is becoming more and more bloated.
``` std::vector<reco::MuonSegmentMatch> segmentMatches ``` can have a value only among dtSegmentRef or cscSegmentRef, while 
```std::vector<reco::MuonSegmentMatch> gemMatches``` can have a meaningful value only for gemSegmentRef
... and inside the MuonChamberMatch only one of these vectors has a meaning.

This is probably still OK considering the most frequent access patterns: actual segments are used mainly in detailed detector analysis (truth matching and alike).
The alternatives will likely lead to polymorphic generalized more effective storage, but, very likely, access will require casting back to specific classes.